File: calc.py
from reader import read
import readline


def repl():
    while True:
        try:
            print(read(input(">")).eval())
        except SyntaxError as e:
            print(type(e).__name__, ":", e)
        except (EOFError, KeyboardInterrupt):
            return


if __name__ == "__main__":
    repl()

File: calc.js
const read = require('./reader');
const readlineSync = require('readline-sync');

function repl() {
  while (true) {
    try {
      console.log(read(readlineSync.question("> ")).eval());
    } catch (e) {
        console.log(e);
    }
  }
}

if (require.main === module) {
  repl();
}
